src/main/kotlin/blackjack/controller/Controller.kt

```diff
@@ -0,0 +1,123 @@
+        players.forEach { it.addCard(getInitialCards()) }
+        OutputView.displayCardsOfDealer(dealer)
+        OutputView.displayCardsOfPlayers(players)
+    }
+
+    private fun finalDraw() {
+        players.forEach { playerTakesTurn(it) }
+        dealerTakesTurn()
+        OutputView.displayCardsOfPlayersWithScore(listOf(dealer) + players)
+        OutputView.displayFinalResultsHeading()
+    }
+
+    private fun getInitialCards(): List<Card> = deck.drawCards(INITIAL_CARD_COUNT)
+
+    private fun dealerTakesTurn() {
+        while (dealer.isDealerBelowMinScore()) {
+            dealer.addCard(deck.drawCards())
+            OutputView.displayDealersTurn()
+        }
+    }
+
+    private fun playerTakesTurn(player: Gambler) {
+        var answer = false
+        while (player.isPlayerBelowBlackJack()) {
+            answer = processHitOrStay(player)
+            if (!answer) {
+                break
+            }
+            player.addCard(deck.drawCards())
+            OutputView.displayCardsOfPlayers(player)
+        }
+        if (!answer && player.cards.size == INITIAL_CARD_COUNT) {
+            OutputView.displayCardsOfPlayers(player)
+        }
+    }
+
+    private fun showResults(finalResult: FinalResult) {
+        OutputView.displayPlayerResult(
+            finalResult.lose.size,
+            finalResult.win.size,
+            finalResult.draw.size,
+        )
+        finalResult.win.forEach {
+            OutputView.displayPlayerResult(it.name, true)
+        }
+        finalResult.lose.forEach {
+            OutputView.displayPlayerResult(it.name, false)
+        }
+        finalResult.draw.forEach {
+            OutputView.displayPlayerResult(it.name, false)
+        }
+    }
+
+    private fun processPlayerNames(): List<GamblerInfo> {
+        repeat(MAX_ATTEMPTS) {
+            try {
+                val names = InputView.getNamesOfPlayers()
+                return names
+                    .parseCommaString()
+                    .map(::GamblerInfo)
+            } catch (err: IllegalArgumentException) {
+                OutputView.displayErrorMessages(err.message)
+            }
+        }
+        throw IllegalArgumentException(MAX_ATTEMPT_MESSAGE)
+    }
+
+    private fun processHitOrStay(player: Player): Boolean {
+        repeat(MAX_ATTEMPTS) {
+            try {
+                return isHitOrStand(InputView.getHitOrStand(player.name))
+            } catch (err: IllegalArgumentException) {
+                OutputView.displayErrorMessages(err.message)
+            }
+        }
+        throw IllegalArgumentException(MAX_ATTEMPT_MESSAGE)
+    }
+
+    private fun isHitOrStand(input: String): Boolean {
+        return when (input) {
+            "y" -> true
+            "n" -> false
+            else -> throw IllegalArgumentException("The answer must be y or n.")
+        }
+    }
+
+    companion object {
+        private const val MAX_ATTEMPTS = 5
+        const val BLACKJACK_SCORE = 21
```

Comment:
If this magic number is only used in `FinalResult`,
we can move this to `FinalResult` and maintain it there.